File: commands/vouch.py
import discord
from discord import app_commands
from discord.ext import commands

# Liste des items disponibles (à mettre depuis ton shop)
ITEMS = [
    "Nitro Boost 1 Month",
    "Nitro Boost 1 Year",
    "Discord Account",
    "Server Boost 14x"
]

# Liste des moyens de paiement autorisés
PAYMENTS = ["PayPal", "LTC"]

class Vouch(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @app_commands.command(name="vouch", description="Donne ton avis sur un service")
    @app_commands.describe(
        vendeur="Mentionne le vendeur",
        quantite="Quantité achetée",
        item="Choisis l'item",
        prix="Prix payé (€ ou crypto)",
        moyen_de_paiement="Moyen de paiement",
        note="Note entre 1 et 5",
        commentaire="Commentaire (facultatif)",
        anonyme="Rendre le vouch anonyme ? Oui / Non"
    )
    @app_commands.choices(
        item=[discord.app_commands.Choice(name=i, value=i) for i in ITEMS],
        moyen_de_paiement=[discord.app_commands.Choice(name=p, value=p) for p in PAYMENTS],
        note=[discord.app_commands.Choice(name=str(i), value=i) for i in range(1, 6)],
        anonyme=[
            discord.app_commands.Choice(name="Oui", value="Oui"),
            discord.app_commands.Choice(name="Non", value="Non")
        ]
    )
    async def vouch(
        self, interaction: discord.Interaction,
        vendeur: discord.Member,
        quantite: int,
        item: str,
        prix: str,
        moyen_de_paiement: str,
        note: int,
        commentaire: str = "Aucun commentaire",
        anonyme: str = "Non"
    ):
        """Slash command /vouch"""

        # Gestion anonymat
        vendeur_display = vendeur.mention if anonyme == "Non" else "👤 Anonyme"

        # Étoiles
        stars = "⭐" * note

        embed = discord.Embed(
            title=f"📝 Nouveau Vouch",
            color=discord.Color.gold()
        )
        embed.add_field(name="👤 Vendeur", value=vendeur_display, inline=True)
        embed.add_field(name="📦 Quantité", value=str(quantite), inline=True)
        embed.add_field(name="🎁 Item", value=item, inline=True)
        embed.add_field(name="💰 Prix", value=prix, inline=True)
        embed.add_field(name="💳 Paiement", value=moyen_de_paiement, inline=True)
        embed.add_field(name="⭐ Note", value=stars, inline=False)
        embed.add_field(name="💬 Commentaire", value=commentaire, inline=False)

        embed.set_footer(text="Système de Vouch • Optimisé")

        await interaction.response.send_message(embed=embed)

async def setup(bot):
    await bot.add_cog(Vouch(bot))

File: commands/vouch.js
const { SlashCommandBuilder, EmbedBuilder } = require("discord.js");
const { addVouch, getVouchCount } = require("../utils/vouches");

module.exports = {
    data: new SlashCommandBuilder()
        .setName("vouch")
        .setDescription("Donne ton avis sur un service")
        .addUserOption(o => o.setName("vendeur").setDescription("Mentionne le vendeur").setRequired(true))
        .addIntegerOption(o => o.setName("quantite").setDescription("Quantité achetée").setRequired(true))
        .addStringOption(o => o.setName("item").setDescription("Choisis l'item").setRequired(true)
            .addChoices(
                { name: "Nitro Boost 1 Month", value: "Nitro Boost 1 Month" },
                { name: "Nitro Basic 1 Month", value: "Nitro Basic 1 Month" },
                { name: "Discord Account", value: "Discord Account" },
                { name: "Server Boost", value: "Server Boost" },
                { name: "Message Réaction", value: "Message Réaction" },
                { name: "Décoration", value: "Décoration" }
            ))
        .addStringOption(o => o.setName("prix").setDescription("Prix payé").setRequired(true))
        .addStringOption(o => o.setName("moyen_de_paiement").setDescription("Moyen de paiement").setRequired(true)
            .addChoices(
                { name: "Paypal", value: "Paypal" },
                { name: "Litecoin", value: "Litecoin" }
            ))
        .addIntegerOption(o => o.setName("note").setDescription("Note 1-5").setRequired(true))
        .addStringOption(o => o.setName("commentaire").setDescription("Commentaire").setRequired(true)),

    async execute(interaction) {
        const vendeur = interaction.options.getUser("vendeur");
        const quantite = interaction.options.getInteger("quantite");
        const item = interaction.options.getString("item");
        const prix = interaction.options.getString("prix");
        const moyen = interaction.options.getString("moyen_de_paiement");
        const note = interaction.options.getInteger("note");
        const commentaire = interaction.options.getString("commentaire");

        const stars = "⭐".repeat(note) + "☆".repeat(5 - note);

        const vouchNumber = getVouchCount() + 1;

        const embed = new EmbedBuilder()
            .setTitle(`📩 Nouveau Vouch`)
            .setColor("#3366FF")
            .setThumbnail(interaction.user.displayAvatarURL({ size: 1024 }))
            .addFields(
                { name: "👤 Vendeur", value: `<@${vendeur.id}>`, inline: false },
                { name: "🛒 Item vendu", value: `${quantite}x ${item} (${prix} via ${moyen})`, inline: false },
                { name: "⭐ Note", value: stars, inline: false },
                { name: "📌 Vouch N°", value: `${vouchNumber}`, inline: false },
                { name: "✍️ Vouch par", value: `<@${interaction.user.id}>`, inline: false },
                { name: "🕒 Date du vouch", value: new Date().toLocaleString("fr-FR"), inline: false },
                { name: "💬 Commentaire", value: commentaire, inline: false }
            )
            .setFooter({ text: "Service proposé par Lightvault by 3keh" });

        const channel = interaction.guild.channels.cache.get("1417943146653810859");
        if (!channel) return interaction.reply({ content: "❌ Channel introuvable.", ephemeral: true });

        const msg = await channel.send({ embeds: [embed] });

        addVouch(interaction.user.id, {
            vendeur_id: vendeur.id,
            quantite,
            item,
            prix,
            moyen,
            note,
            commentaire,
            date: new Date().toISOString(),
            messageId: msg.id
        });

        await interaction.reply({ content: "✅ Ton vouch a été envoyé !", flags: 64 });
    }
};
